serverless/serverless
# Implementing Error tests (per issue #3169)

## What did you implement:

Closes #3169 - Error tests implementation

## How did you implement it:



I was checking this Serverless framework you folks are developing, and it looked interesting. So I've decided to spend some time contributing to it. For the initial PR I've decided to start with something simple - like covering a class with tests.

`lib/classes/Error.js` was not covered - so it was a good thing to start with.

Note, that this PR contains the following changes:

1. Error coverage - major change, pretty straightforward.
2. For the `Error.js` export I renamed `SError` export to `ServerlessError` (the backwards compatibility was left, though). A couple of reasons for that. First, it brought back the correct class name right to the export. Second - searching source code by string (for refactoring or just to find class usage) now will return all the places where the class is directly imported.
3. Changed "construtor" to "constructor" - a trivial typo fix. It was almost the first thing I saw in tests log, which was shouting to have it fixed :)

## How can we verify it:

Run testing and check the coverage:
```
npm test
```

Or verify my [Travis](https://travis-ci.org/zerkella/serverless/builds/196879631) and [Coveralls](https://coveralls.io/builds/9919985) results.

## Todos:

- [ ] N/A - Write documentation

***Is this ready for review?:*** YES
***Is it a breaking change?:*** NO
